src/box/space_cache.c

```diff
@@ -54,6 +54,8 @@ space_cache_destroy(void)
 struct space *
 space_by_id(uint32_t id)
 {
+	if (spaces == NULL)
+		return NULL;
```

Comment:
Instead of adding the check to a relatively hot-path function, better initialize the hash tables in `box_init()`.

Reply:
1. Now there are two calls to `space_cache_init`, which is wrong. You need to delete the old one.
2. `box_init` is matched by `box_free` so you should also move `space_cache_destroy`. Since the `schema_free` is commented out, it's better to leave `space_cache_destroy` commented out as well.

<details><summary>diff</summary>

```diff
diff --git a/src/box/box.cc b/src/box/box.cc
index ad0fffc09587..9ecf13706efc 100644
--- a/src/box/box.cc
+++ b/src/box/box.cc
@@ -5099,6 +5099,7 @@ box_init(void)
 	fiber_cond_create(&ro_cond);
 	auth_init();
 	security_init();
+	space_cache_init();
 	user_cache_init();
 	/*
 	 * The order is important: to initialize sessions, we need to access the
@@ -5138,4 +5139,5 @@ box_free(void)
 	/* schema_module_free(); */
 	/* session_free(); */
 	/* user_cache_free(); */
+	/* space_cache_destroy(); */
 }
diff --git a/src/box/schema.cc b/src/box/schema.cc
index 2cb02f191849..94a487b85574 100644
--- a/src/box/schema.cc
+++ b/src/box/schema.cc
@@ -205,8 +205,6 @@ schema_init(void)
 	for (uint32_t i = 0; i < lengthof(key_parts); i++)
 		key_parts[i] = key_part_def_default;
 
-	/* Initialize the space cache. */
-	space_cache_init();
 	func_cache_init();
 	sequences = mh_i32ptr_new();
 	/*
@@ -348,7 +346,6 @@ schema_init(void)
 void
 schema_free(void)
 {
-	space_cache_destroy();
 	func_cache_destroy();
 
 	while (mh_size(sequences) > 0) {
diff --git a/src/box/space_cache.c b/src/box/space_cache.c
index 88f1dce22f50..cddcdadc23bf 100644
--- a/src/box/space_cache.c
+++ b/src/box/space_cache.c
@@ -36,8 +36,6 @@ space_cache_init(void)
 void
 space_cache_destroy(void)
 {
-	if (spaces == NULL)
-		return;
 	while (mh_size(spaces) > 0) {
 		mh_int_t i = mh_first(spaces);
 
```

</details>